fabric-tool-attribute-api-v1/src/main/java/net/fabricmc/fabric/impl/tool/attribute/handlers/TaggedToolsModdedBlocksToolHandler.java

```diff
@@ -41,13 +42,13 @@
 	@NotNull
 	@Override
 	public ActionResult isEffectiveOn(Tag<Item> tag, BlockState state, ItemStack stack, @Nullable LivingEntity user) {
-		if (!(stack.getItem() instanceof DynamicAttributeTool) && !(stack.getItem() instanceof ToolItem)) {
+		if (!(stack.getItem() instanceof FabricTool)) {
 			@Nullable ToolManagerImpl.Entry entry = ToolManagerImpl.entryNullable(state.getBlock());
 
 			if (entry != null) {
 				int miningLevel = 0; // minimum mining level: the tool is tagged but nothing else is said about it
 				int requiredMiningLevel = entry.getMiningLevel(tag);
-				return requiredMiningLevel >= miningLevel ? ActionResult.SUCCESS : ActionResult.PASS;
+				return miningLevel >= requiredMiningLevel ? ActionResult.SUCCESS : ActionResult.PASS;
```

Comment:
My fix for this is a bit more comprehensive (and you need to fix the merge conflicts anyway).

Also, are all `ToolItem`s `FabricTool`s? If not, the if check above would have a different meaning

Reply:
Did I comment just as you fixed the conflicts? :eyes:

Reply:
Haha, yeah I fixed the conflicts pretty quickly.

And yes, `ToolItem` implements `FabricTool` courtesy of `ToolItemMixin`. All `ToolItem`s, `ShearsItem`s, and `DynamicAttributeTool`s are `FabricTool`s.